torch/_dynamo/exc.py

```diff
@@ -182,11 +182,13 @@ def unimplemented_with_warning(e: Exception, code, msg: str) -> NoReturn:
     graph_break_msg = format_error_msg_verbose(e, code)
     graph_breaks_log.debug("%s", graph_break_msg)
     log.warning(msg)
-    raise unimplemented(msg) from e
+    unimplemented(msg, from_exc=e)
 
 
-def unimplemented(msg: str) -> NoReturn:
+def unimplemented(msg: str, *, from_exc: Optional[Exception] = None) -> NoReturn:
     assert msg != os.environ.get("BREAK", False)
+    if from_exc is not None:
```

Comment:
What if people want to raise from None to suppress the automatic exception chaining? You might need to use a sentinel value here instead.